ccm_web/server/src/api/api.admin.handler.ts

```diff
@@ -102,8 +119,12 @@ export class AdminApiHandler {
     const result = makeResponse<CanvasCourse[]>(coursesResponses)
     if (isAPIErrorData(result)) return result
     const allCourses: CanvasCourse[] = []
-    result.map(cs => allCourses.push(...cs))
+    result.forEach(cs => allCourses.push(...cs))
     logger.debug(`Number of courses matching search term: ${allCourses.length}`)
+    if (allCourses.length > this.maxSearchCourses) {
+      this.logTooManyCoursesErrorDetails(queryParams)
+      throw new TooManyResultsError()
```

Comment:
I found a perfect user who belongs to multiple sub-account, but I couldn't trigger this case, mostly from the former call the TooManyResultError is triggered. But sometime you can't trigger all cases I will go by logic here.